File: php/src/Model/AttachmentDownload.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A short-lived download URL.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class AttachmentDownload
{
    public function __construct(
        public readonly string $url,
        public readonly int $expiresAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            url: (string) ($data['url'] ?? ''),
            expiresAt: (int) ($data['expiresAt'] ?? 0),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/AttachmentDownload.java
package io.fivexer.sdk.model;

/**
 * A short-lived download URL.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class AttachmentDownload {
    private String url;
    private long expiresAt;

    public String getUrl() { return url; }
    public long getExpiresAt() { return expiresAt; }
}
